DQMOffline/MuonDPG/plugins/GEMTnPEfficiencyTask.cc

```diff
@@ -0,0 +1,848 @@
+  BaseTnPEfficiencyTask::analyze(event, context);
+
+  edm::Handle<reco::MuonCollection> muons;
+  event.getByToken(m_muToken, muons);
+
+  //GEM variables
+  std::vector<std::vector<int>> probe_coll_GEM_region;
+  std::vector<std::vector<int>> probe_coll_GEM_lay;
+  std::vector<std::vector<int>> probe_coll_GEM_chamber;
+  std::vector<std::vector<float>> probe_coll_GEM_pt;
+  std::vector<std::vector<float>> probe_coll_GEM_eta;
+  std::vector<std::vector<float>> probe_coll_GEM_phi;
+  std::vector<std::vector<int>> probe_coll_GEM_sta;
+  std::vector<std::vector<float>> probe_coll_GEM_dx;
+  std::vector<uint8_t> probe_coll_GEM_staMatch;
+
+  //ME0 variables
+  std::vector<std::vector<int>> probe_coll_ME0_region;
+  std::vector<std::vector<int>> probe_coll_ME0_roll;
+  std::vector<std::vector<int>> probe_coll_ME0_lay;
+  std::vector<std::vector<int>> probe_coll_ME0_chamber;
+  std::vector<std::vector<float>> probe_coll_ME0_pt;
+  std::vector<std::vector<float>> probe_coll_ME0_eta;
+  std::vector<std::vector<float>> probe_coll_ME0_phi;
+  std::vector<std::vector<int>> probe_coll_ME0_sta;
+  std::vector<std::vector<float>> probe_coll_ME0_dx;
+  std::vector<uint8_t> probe_coll_ME0_staMatch;
+
+  std::vector<unsigned> probe_indices;
+  if (!m_probeIndices.empty())
+    probe_indices = m_probeIndices.back();
+
+  //Fill probe dx + subdetector coordinates
+  for (const auto i : probe_indices) {
+    //GEM variables
+    std::vector<int> probe_GEM_region;
+    std::vector<int> probe_GEM_sta;
+    std::vector<int> probe_GEM_lay;
+    std::vector<int> probe_GEM_chamber;
+    std::vector<float> probe_GEM_pt;
+    std::vector<float> probe_GEM_eta;
+    std::vector<float> probe_GEM_phi;
+    std::vector<float> probe_GEM_dx;
+    //std::vector<float> probe_GEM_dx_seg;
+    uint8_t GEM_stationMatching = 0;
+    //ME0 variables
+    std::vector<int> probe_ME0_region;
+    std::vector<int> probe_ME0_roll;
+    std::vector<int> probe_ME0_sta;
+    std::vector<int> probe_ME0_lay;
+    std::vector<int> probe_ME0_chamber;
+    std::vector<float> probe_ME0_pt;
+    std::vector<float> probe_ME0_eta;
+    std::vector<float> probe_ME0_phi;
+    std::vector<float> probe_ME0_dx;
+    uint8_t ME0_stationMatching = 0;
+
+    float gem_matched = false;  // fill detailed plots only for probes matching GEM
+
+    for (const auto& chambMatch : (*muons).at(i).matches()) {
+      // look in GEMs
+      if (chambMatch.detector() == MuonSubdetId::GEM) {
+        if (chambMatch.edgeX < m_borderCut && chambMatch.edgeY < m_borderCut) {
+          gem_matched = true;  //fill detailed plots if at least one GEM match
+
+          GEMDetId chId(chambMatch.id.rawId());
+
+          int roll = chId.roll();
+          int region = chId.region();
+          int station = chId.station();
+          int layer = chId.layer();
+          int chamber = chId.chamber();
+          float pt = (*muons).at(i).pt();
+          float eta = (*muons).at(i).eta();
+          float phi = (*muons).at(i).phi();
+
```

Comment:
I think they can all be made `const` (but please check)
```suggestion
          const int roll = chId.roll();
          const int region = chId.region();
          const int station = chId.station();
          const int layer = chId.layer();
          const int chamber = chId.chamber();
          const float pt = (*muons).at(i).pt();
          const float eta = (*muons).at(i).eta();
          const float phi = (*muons).at(i).phi();

```